### PR TITLE
set default mllm dataset

### PR DESCRIPTION
**Pull request by n1ck-guo**

### CODE REVIEW

**Comment by wenhuach21** [comment]
should add a extra_requirement.txt, right?

File: auto_round/autoround.py [hunk 1]
```diff
@@ -579,7 +579,7 @@ def calib(self, nsamples, bs):
         elif total_cnt < nsamples:
             logger.warning(
                 f"Insufficient number of samples collected may affect the quantification. "
-                f"Valid samples size:{total_cnt}, Target sample size:{nsamples}"
+                f"Valid samples size {total_cnt}, while target sample size is only {nsamples}"
```
Comment (wenhuach21) [inline auto_round/autoround.py]: target samples count is  {nsamples}, while valid samples count is  {total_cnt}

File: auto_round/mllm/autoround_mllm.py [hunk 2]
```diff
@@ -172,6 +186,18 @@ def __init__(
             **kwargs,
         )
 
+        
+    def _only_text_test(self):
+        """Test if the model whether can use text-only datasets."""
+        try:
+            text =  ["only text test", "whether can use text-only dataset"]
```
Comment (wenhuach21) [inline auto_round/mllm/autoround_mllm.py]: make it shorter, as large model will on cpu

File: auto_round/mllm/mllm_dataset.py [hunk 3]
```diff
@@ -67,15 +68,20 @@ def __init__(
...
             extra_data_dir=None,
-            seqlen=None,
+            seqlen=512,
             padding=True,
             truncation=True,
+            nsamples=512
             ) -> None:
         super().__init__()
         self.model = model
         self.model_type = template.model_type
```
Comment (wenhuach21) [inline auto_round/mllm/mllm_dataset.py:79]: where is the code to set it to 512 even users set it to 2048

Comment (n1ck-guo) [reply auto_round/mllm/mllm_dataset.py:79]: in mllm_dataset.py line 233

File: auto_round/mllm/mllm_dataset.py [hunk 4]
```diff
@@ -230,7 +230,7 @@ def get_mllm_dataloader(
     if os.path.isfile(dataset) or dataset in MLLM_DATASET.keys():
         dataset = MLLM_DATASET['liuhaotian/llava'](
             template, model, tokenizer, dataset, extra_data_dir, 
-            seqlen=min(seqlen, 111512), truncation=truncation, nsamples=nsamples)
+            seqlen=min(seqlen, 512), truncation=truncation, nsamples=nsamples)
```
Comment (wenhuach21) [inline auto_round/mllm/mllm_dataset.py]: better not hard coded, add a value

File: auto_round/mllm/autoround_mllm.py [hunk 5]
```diff
@@ -130,8 +130,22 @@ def __init__(
...
             self.template, model=model, tokenizer=tokenizer, image_processor=image_processor)
+        
+        if dataset is None:
+            dataset = self.template.default_dataset
+        if truncation is None:
+            truncation = True if dataset == "NeelNanda/pile-10k" else False
         self.truncation = truncation
-        assert dataset is not None, "dataset should not be None"
+
+        if not self._only_text_test() and dataset == "NeelNanda/pile-10k":
```
Comment (WeiweiZhang1) [inline auto_round/mllm/autoround_mllm.py]: better change this string to register llm dataset dict

File: auto_round/mllm/mllm_dataset.py [hunk 6]
```diff
@@ -103,19 +110,33 @@ def __init__(
...
+                    new_questions.append(source)
+            if len(new_questions) >= nsamples:
+                return new_questions
+            else:
+                if seqlen > max_len:
+                    logger.error(f"seqlen={seqlen} is greater than the max length of dataset {max_len},"
+                                 f" will change seqlen to {max_len - 128}")
+                    new_min_seqlen = max_len - 128
+                else:
+                    logger.error(f"not enough sample for seqlen greater than {min_seqlen},"
```
Comment (WeiweiZhang1) [inline auto_round/mllm/mllm_dataset.py]: no enough samples ?

File: auto_round/script/mllm.py [hunk 7]
```diff
@@ -197,13 +199,13 @@ def setup_parser():
...
 
-    parser.add_argument("--batch_size", "--train_bs", default=1, type=int,
+    parser.add_argument("--batch_size", "--train_bs", default=8, type=int,
                         help="train batch size")
 
     parser.add_argument("--iters", "--iter", default=200, type=int,
                         help=" iters")
 
-    parser.add_argument("--seqlen", "--seq_len", default=256, type=int,
+    parser.add_argument("--seqlen", "--seq_len", default=1024, type=int,
```
Comment (WeiweiZhang1) [inline auto_round/script/mllm.py]: default is not 2048 for pile?

**Review by WeiweiZhang1 — APPROVED** [review]

**Pull request merged**